### PR TITLE
Add `whenViewHidden` for debug status in status bar

### LINKED ISSUE #40212: Status bar debug launcher should only be visible when debug view is closed
It seems daft to fill up with status bar with another widget when the configuration selector and start button are already available.

### PR DESCRIPTION
**Pull request by Krzysztof-Cieslak**
Fix #40212

CC: @isidorn

### CODE REVIEW

**Comment by isidorn** [comment]
@Krzysztof-Cieslak this PR has conficlits with the main branch, can you please resolve conflicts and then I can review in more detail. I also left some initial comments directly in the code.

Thanks!

File: src/vs/workbench/parts/debug/browser/debugStatus.ts [hunk 1]
```diff
@@ -24,12 +25,14 @@ export class DebugStatus extends Themable implements IStatusbarItem {
 	private label: HTMLElement;
 	private icon: HTMLElement;
 	private showInStatusBar: string;
```
Comment (isidorn) [inline src/vs/workbench/parts/debug/browser/debugStatus.ts]: showInStatusBar should be a ''never' | 'always' | 'onFirstSessionStart' | 'whenViewHidden';

File: src/vs/workbench/parts/debug/browser/debugStatus.ts [hunk 2]
```diff
@@ -47,7 +50,13 @@ export class DebugStatus extends Themable implements IStatusbarItem {
 				this.showInStatusBar = configurationService.getValue<IDebugConfiguration>('debug').showInStatusBar;
 				if (this.showInStatusBar === 'never' && this.statusBarItem) {
 					this.statusBarItem.hidden = true;
-				} else {
+				}
```
Comment (isidorn) [inline src/vs/workbench/parts/debug/browser/debugStatus.ts]: can you put the `else` branch on the same line with the closing bracket. I just personally prefer it that way

File: src/vs/workbench/parts/debug/browser/debugStatus.ts [hunk 3]
```diff
@@ -24,12 +25,14 @@ export class DebugStatus extends Themable implements IStatusbarItem {
 	private label: HTMLElement;
 	private icon: HTMLElement;
 	private showInStatusBar: string;
+	private viewletId: string;
```
Comment (isidorn) [inline src/vs/workbench/parts/debug/browser/debugStatus.ts]: I believe `this.viewletId` is not needed

File: src/vs/workbench/parts/debug/browser/debugStatus.ts [hunk 4]
```diff
@@ -57,6 +66,29 @@ export class DebugStatus extends Themable implements IStatusbarItem {
...
 			}
 		}));
+
+		this.toDispose.push(viewletService.onDidViewletOpen(e => {
+			this.viewletId = e.getId();
+			if (this.showInStatusBar === 'whenViewHidden') {
+				//Show status bar item if new view is not debug
+				if (this.viewletId !== VIEWLET_ID) {
+					if (!this.statusBarItem) {
+						this.doRender();
```
Comment (isidorn) [inline src/vs/workbench/parts/debug/browser/debugStatus.ts]: No need for this case imho. The onDidViewletClose will catch it.
You should just check if openeing a viewlet shuold hide the statusbar

**Comment by Krzysztof-Cieslak** [comment]
Sorry for delay, I've been bit busy over last couple of weeks. I'll take a look at it tomorrow.

**Comment by Krzysztof-Cieslak** [comment]
@isidorn, this should be OK now.

File: src/vs/workbench/parts/debug/browser/debugStatus.ts (hunk 3 above)
Comment (Krzysztof-Cieslak) [reply src/vs/workbench/parts/debug/browser/debugStatus.ts]: It's required for handling configuration change correctly - we need to know if we are in Debug Viewlet when someone changed configuration to hide the status bar item if necessary.

**Pull request closed without merging**